spring-cloud-dataflow-registry/src/main/java/org/springframework/cloud/dataflow/registry/repository/AppRegistrationRepositoryImpl.java

```diff
@@ -91,17 +92,11 @@ public Page<AppRegistration> findAllByTypeAndNameIsLikeAndVersionAndDefaultVersi
 				appRegistration.setVersions(versions);
 			});
 		}
-		return new PageImpl<>(resultList, pageable, getTotalCount(cb, predicates.toArray(new Predicate[0])));
+		return new PageImpl<>(resultList, pageable, getTotalCount(cq));
 	}
 
-	private Long getTotalCount(CriteriaBuilder criteriaBuilder, Predicate[] predicateArray) {
-		CriteriaQuery<Long> criteriaQuery = criteriaBuilder.createQuery(Long.class);
-		Root<AppRegistration> root = criteriaQuery.from(AppRegistration.class);
-
-		criteriaQuery.select(criteriaBuilder.count(root));
-		criteriaQuery.where(predicateArray);
-
-		return entityManager.createQuery(criteriaQuery).getSingleResult();
+	private Long getTotalCount(CriteriaQuery<AppRegistration> criteriaQuery) {
+		return (Long) entityManager.createQuery(((SqmSelectStatement)criteriaQuery).createCountQuery()).getSingleResult();
```

Comment:
Is `criteraQuery` always a `SqmSelectStatement` ?

Reply:
for these two statements, yes.